src/iris/sender/coordinator.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from iris import db, metrics
+from gevent import sleep, spawn
+from itertools import cycle
+
+logger = logging.getLogger(__name__)
+
+UPDATE_FREQUENCY = 3
+SENDER_ALIVE_TIMEOUT = 10
+
+REMOVE_OLD_INSTANCES_FREQUENCY = 3600
+REMOVE_OLD_INSTANCES_TIMEOUT = 60
+
+GET_MASTER_QUERY = '''SELECT `sender_address` FROM `sender_master_election` WHERE `anchor` = 1'''
+
+GET_SLAVES_QUERY = '''
+  SELECT `sender_address`
+  FROM `sender_instances`
+  WHERE `last_seen` > NOW() - INTERVAL CAST(:timeout AS UNSIGNED) SECOND
+  AND `sender_address` NOT IN (%s)''' % GET_MASTER_QUERY
+
+UPDATE_MASTER_QUERY = '''
```

Comment:
I am interested in how well this tolerates the Galera WAN cluster - it looks potentially deadlocky especially in the case of clock drift on the DB nodes.